src/applications/random.jl

```diff
@@ -0,0 +1,38 @@
+Random.gentype(::Type{<:Domain{T}}) where T = T
+
+Base.rand(rng::AbstractRNG, s::Random.SamplerTrivial{<:ProductDomain}) = toexternalpoint(s[], map(i->(rand(rng, i)), factors(s[])))
+
+Base.rand(rng::AbstractRNG, s::Random.SamplerTrivial{<:SimpleLazyDomain}) = toexternalpoint(s[], rand(rng, superdomain(s[])))
+
+function Base.rand(rng::AbstractRNG, s::Random.SamplerTrivial{<:Ball})
+    # Technical details: http://extremelearning.com.au/how-to-generate-uniformly-random-points-on-n-spheres-and-n-balls/
+
+    b = s[]
+    
+    # for low-dimensional balls, use rejection sampling - acceptance rate is at least 52%
+    if dimension(b) <= 3
+        while true
+            r = rand(rng, boundingbox(b))
```

Comment:
`boundingbox(d)` allocates memory and is not very optimized, perhaps it is better to define `box=boundingbox(b)` once before the while loop?

Reply:
> boundingbox(d) allocates memory and is not very optimized

allocates memory even with StaticArrays? that is unfortunate.